client/app/components/packages/attachments.js

```diff
@@ -0,0 +1,21 @@
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+
+/**
+  * This component creates a new File Manager specifically for
+  * the passed in package. The File Manager is among one of many
+  * on the the application File Manager service.
+  * @param      {Package Model}  package
+  */
+export default class PackagesAttachmentsComponent extends Component {
+  @service
+  fileManagement;
+
+  get fileManager() {
+    return this.fileManagement.fileManagers[this.args.package.id]
+    || this.fileManagement.createFileManager(
+      this.args.package.id,
+      this.args.package.documents,
+    );
+  }
```

Comment:
Hmm, so I recall running into some trouble with this approach in LUP portal. Do you recall?

In my experience, it can get tricky when you _create_ things inside of a _getter_ (or a computed property). The reason for that is it's a side-effect of calling the property (`myComponent.fileManager`): your native getter there, which should just _read_ some data from somewhere, actually _may_ conditionally create something.

Alternatively, could we create/lookup a file manager somewhere else and pass it into this component?

I'm guessing that the reasoning here is that we could return to the component, and we want a way to retain the same state. But I wonder if the way to handle that is to figure out what happens when the user navigates away from the form, and un-uploaded files are still in-memory. What if the component clears the file manager when it's destroyed?

Reply:
Hmm at this point whether navigating away from the component clears the files is an open design issue I think. But I do agree it would be better if the File Manager and the file references aren't destroyed when navigating away. If component teardown did somehow clear the file manager,  existing and to-be-deleted file references would be lost. The to-be-uploaded file queue should still remain in memory. 

I think the same would go for if File Managers sat on the Package model, and the model teardown clears the File Manager

Reply:
Why should it remain in memory? Wouldn't that risk memory leakage? We experienced this issue during the LUP build, I think we should try to avoid it, and that's ultimately what I'm suggesting here. Component teardown wouldn't automatically clear the files, we'd have to wire that up correctly.

Either way, I think we still should not have a component that implicitly (and conditionally) creates things in memory. I think it's a confusing design because it would surprise me if I read a property from an object, and the act of observing/reading creates things in memory.

Reply:
Oh, I didn't meant that part as a prescriptive statement. I meant that if a FileManager is deleted on teardown, it doesn't necessarily also delete the referenced fileQueue. We can for sure manually destroy the fileQueue too. 

Originally thought that memory leakage shouldn't be a big issue if there's only 10-15 editable packages on the dashboard at any given time. 

Though now that I'm thinking about it, if somehow the browser cache holds on to File Managers and File Queues indefinitely (because the FileManagement service is independent of loaded packages), I see how it will leak over months/years. Hmm

Reply:
Okay, I get now why it may be ideal to create the File Manager+fileQueue on Package model init and destroy both on model didDelete()

Reply:
Ah, gotcha! Makes sense :) 

Memory leakage to me in this situation means something like file uploads somehow persisting in-memory across different packages, even if they were navigate away from. It's not that it's a big deal in terms of performance, but it could confuse us from a maintainability perspective.